src/redux/services/templates.ts

```diff
@@ -128,24 +130,33 @@ export const createUnsavedResourcesFromVanillaTemplate = async (
       }
     }
   );
-  const createdResources: K8sResource[] = [];
+
+  let objects: any[] = [];
+
   resourceTextList
     .filter((text): text is string => typeof text === 'string')
     .forEach(resourceText => {
-      const objects = extractObjectsFromYaml(resourceText);
-      objects.forEach(obj => {
-        const resource = createUnsavedResource(
-          {
-            name: obj.metadata.name,
-            namespace: obj.metadata.namespace,
-            kind: obj.kind,
-            apiVersion: obj.apiVersion,
-          },
-          dispatch,
-          obj
-        );
-        createdResources.push(resource);
-      });
+      objects = [...objects, ...extractObjectsFromYaml(resourceText)];
     });
+
+  const createdResources: K8sResource[] = [];
+
+  for (const obj of objects) {
+    // eslint-disable-next-line no-await-in-loop
+    await sleep(500);
```

Comment:
Why do we have a timeout here?

Reply:
Because The state value gets overwritten if we call 2`createResource` functions immediately.

Reply:
I'd like to test this for some time to see if it can be fixed in another way.
I think even a new action that creates multiple unsaved resources at once could be better... we should avoid relying on timeouts like this one